src/nxt_var.c

```diff
@@ -433,11 +433,18 @@ nxt_var_query_init(nxt_var_query_t **query_p, void *ctx, nxt_mp_t *mp)
             return NXT_ERROR;
         }
 
-        nxt_array_init(&query->values, mp, sizeof(nxt_var_value_t));
-        nxt_array_init(&query->parts, mp, sizeof(nxt_str_t *));
+        query->values = nxt_array_create(mp, 1, sizeof(nxt_var_value_t));
+        if (nxt_slow_path(query->values == NULL)) {
+            return NXT_ERROR;
+        }
+
+        query->parts = nxt_array_create(mp, 1, sizeof(nxt_str_t *));
+        if (nxt_slow_path(query->parts == NULL)) {
+            return NXT_ERROR;
+        }
```

Comment:
This results in 4 additional allocations in the most common case.

Reply:
I agree, I underestimated the design of the project in this place, and also did not know about the high cost of calls